experimental/iterators/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+cmake_minimum_required(VERSION 3.0.0)
+project(iterators VERSION 0.1.0)
+
+set(CMAKE_CXX_STANDARD 17)
+
+# Set up GoogleTest dependency
+include(FetchContent)
+FetchContent_Declare(
```

Comment:
Traditionally I'd have gone for a git submodule but this seems simple and fine for experimental.

One thing to note is that in the future we will want to impl. tests using FileCheck (https://llvm.org/docs/CommandGuide/FileCheck.html). 

See for example this somewhat complex test (https://github.com/llvm/llvm-project/blob/main/mlir/test/Integration/Dialect/Linalg/CPU/benchmark_matmul.mlir#L98) which starts from MLIR, calls into C / C++ to verify and print 0 on success, then performs checks with FileCheck (https://github.com/llvm/llvm-project/blob/main/mlir/test/Integration/Dialect/Linalg/CPU/benchmark_matmul.mlir#L91).

No need to change anything here, just mentioning so you can think ahead in terms compatible with the current best practices.

Reply:
I had skimmed the repo for tests and found the rewrite tests I think you mention. However, what I am testing currently doesn't seem to apply. It sure does once I get to the MLIR dialect, so it's good to have the pointers. Maybe we can discuss options later online.

Reply:
sg!